python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import json
+import logging
+from typing import List, Optional, Union
+
+import aiohttp
+from pydantic import HttpUrl
+
+from semantic_kernel.connectors.ai.ai_request_settings import AIRequestSettings
+from semantic_kernel.connectors.ai.ai_service_client_base import AIServiceClientBase
+from semantic_kernel.connectors.ai.ollama.ollama_request_settings import OllamaTextRequestSettings
+from semantic_kernel.connectors.ai.text_completion_client_base import (
+    TextCompletionClientBase,
+)
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class OllamaTextCompletion(TextCompletionClientBase, AIServiceClientBase):
+    """
+    Initializes a new instance of the OllamaTextCompletion class.
+
+    Make sure to have the ollama service running either locally or remotely.
+
+    Arguments:
+        ai_model_id {str} -- Ollama model name, see https://ollama.ai/library
+        url {Optional[Union[str, HttpUrl]]} -- URL of the Ollama server, defaults to http://localhost:11434/api/generate
+    """
+
+    url: HttpUrl = "http://localhost:11434/api/generate"
+    session: Optional[aiohttp.ClientSession] = None
+
+    async def complete_async(
+        self,
+        prompt: str,
+        request_settings: OllamaTextRequestSettings,
+        **kwargs,
+    ) -> Union[str, List[str]]:
+        request_settings.prompt = prompt
+        request_settings.stream = False
+        if self.session:
+            async with self.session.post(self.url, json=request_settings.prepare_settings_dict()) as response:
+                response.raise_for_status()
+                return await response.text()
+        async with aiohttp.ClientSession() as session:
+            async with session.post(self.url, json=request_settings.prepare_settings_dict()) as response:
+                response.raise_for_status()
+                return await response.text()
```

Comment:
Same as previous comments, maybe refactor with a `AsyncSession` context manager?

Reply:
I remember seeing this changes and now they are gone and the duplicated logic is back in.